tools/export/sw4stm32/__init__.py

```diff
@@ -79,32 +81,103 @@ def __gen_dir(self, dirname):
     def __generate_uid(self):
         return "%0.9u" % randint(0, 999999999)
 
+    @staticmethod
+    def filter_dot(str):
+        """
+        This function removes ./ from str.
+        str must be converted with win_to_unix() before using this function.
+        """
+	if str == None:
```

Comment:
There seems to be a tabulation here. Python conventions (PEP8) prefer 4 spaces for indentation, and dislike mixed tabs and spaces. Moreover, `str` is a builtin class and should not be used as a variable name: prefer a name like `path` which is also more descriptive of what the variable holds. Also the prefered method to test for `None` is `if path is None` because None is a singleton (also PEP8).